exwm-xsettings.el

```diff
@@ -0,0 +1,343 @@
+  (with-slots (owner)
+      (xcb:+request-unchecked+reply exwm-xsettings--connection
+          (make-instance 'xcb:GetSelectionOwner
+                         :selection exwm-xsettings--XSETTINGS_S0-atom))
+    (when (/= owner xcb:Window:None)
+      (xcb:disconnect exwm-xsettings--connection)
+      (setq exwm-xsettings--connection nil)
+      (warn "[EXWM] Other XSETTINGS manager detected")
+      (cl-return-from exwm-xsettings--init)))
+
+
+  (let ((id(xcb:generate-id exwm-xsettings--connection)))
+    (setq exwm-xsettings--selection-owner-window id)
+
+    ;; Create a settings window.
+    (xcb:+request exwm-xsettings--connection
+        (make-instance 'xcb:CreateWindow
+                       :wid id
+                       :parent exwm--root
+                       :class xcb:WindowClass:InputOnly
+                       :x 0
+                       :y 0
+                       :width 1
+                       :height 1
+                       :border-width 0
+                       :depth 0
+                       :visual 0
+                       :value-mask xcb:CW:OverrideRedirect
+                       :override-redirect 1))
+
+    ;; Set _NET_WM_NAME.
+    (xcb:+request exwm-xsettings--connection
+        (make-instance 'xcb:ewmh:set-_NET_WM_NAME
+                       :window id
+                       :data "EXWM: exwm-xsettings--selection-owner-window"))
+
+    ;; Apply the XSETTINGS properties.
+    (exwm-xsettings--update-settings)
+
+    ;; Take ownership and notify.
+    (xcb:+request exwm-xsettings--connection
+        (make-instance 'xcb:SetSelectionOwner
+                       :owner id
+                       :selection exwm-xsettings--XSETTINGS_S0-atom
+                       :time xcb:Time:CurrentTime))
+    (xcb:+request exwm-xsettings--connection
+        (make-instance 'xcb:SendEvent
+                       :propagate 0
+                       :destination exwm--root
+                       :event-mask xcb:EventMask:StructureNotify
+                       :event (xcb:marshal
+                               ;; TODO: new event? Do I need a new field?
+                               (make-instance 'xcb:systemtray:-ClientMessage
+                                              :window exwm--root
+                                              :time xcb:Time:CurrentTime
+                                              :selection exwm-xsettings--XSETTINGS_S0-atom
+                                              :owner id)
+                               exwm-xsettings--connection)))
+
+    ;; Detect loss of XSETTINGS ownership.
+    (xcb:+event exwm-xsettings--connection 'xcb:SelectionClear
+                #'exwm-xsettings--on-SelectionClear)
+
+    (xcb:flush exwm-xsettings--connection))
+
+  ;; Update the xsettings if/when the theme changes.
+  (add-hook 'enable-theme-functions #'exwm-xsettings--on-theme-change)
+  (add-hook 'disable-theme-functions #'exwm-xsettings--on-theme-change))
```

Comment:
I suggest to provide a public reload function which could be called from the user configuration when the xsettings variables settings are changed instead of registering these hooks.

Reply:
A user should be able to just call `(setopt exwm-xsettings-theme "NewTheme")`, they don't need to call these functions.

You're right that this code isn't, strictly speaking, necessary. Nor is it sufficient for all use-cases. But it probably covers _most_ cases and it's pretty convenient.

Reply:
> A user should be able to just call (setopt exwm-xsettings-theme "NewTheme"), they don't need to call these functions.

Right. Exposing the update function is not necessary.

> You're right that this code isn't, strictly speaking, necessary. Nor is it sufficient for all use-cases. But it probably covers most cases and it's pretty convenient.

Yes, having this functionality here with the possibility to opt-out should not hurt. Maybe if the `exwm-xsettings-theme` variables are not set, the functionality could be completely disabled, no hooks installed?

Reply:
>  the functionality could be completely disabled, no hooks installed?

I considered it but instead made them no-ops when not required.

- The logic required for automatically adding-removing the hooks is more complex and the cost of running the hooks when they're "not used" is minimal.
- Adding them in `exwm-xsettings--init` gives the user a way to reliably remove them if they _really_ don't want this behavior.
- EXWM already hooks into `enable-theme-functions` and `disable-theme-functions` in a few places to detect changes in the background color (system tray and the background module).

Reply:
Thanks! Makes sense. Looking forward to your PR getting merged.